com.unity.ml-agents.extensions/Runtime/Sensors/RigidBodyPoseExtractor.cs

```diff
@@ -105,6 +107,18 @@ public RigidBodyPoseExtractor(Rigidbody rootBody, GameObject rootGameObject = nu
 
             // By default, ignore the root
             SetPoseEnabled(0, false);
+
+            if (enableBodyPoses != null)
```

Comment:
This is so that when we reset the PoseExtractor on the SensorComponent, we keep the existing toggle state of the bodies.